types/src/error.rs

```diff
@@ -2,7 +2,96 @@
 // Parts of the project are originally copyright © Meta Platforms, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-//! Error codes that follow the Move error convention of the Aptos Framework.
+use move_binary_format::errors::PartialVMError;
+use move_core_types::vm_status::StatusCode;
+use std::fmt::Display;
+
+/// Wrapping other errors, to add a variant that represents something that should never
+/// happen - i.e. a code invariant error, which we would generally just panic, but since
+/// we are inside of the VM, we cannot do that.
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PanicError {
```

Comment:
this still cannot be called from third party, right? so we still have duplication?

can there be a separate create that third party can call?

Reply:
I don't think anything is/can be shared in that direction, and defining more general things in third party than needed there also didn't make much sense.